src/lib/schedule.ts

```diff
@@ -0,0 +1,37 @@
+import type { Handler, HandlerContext, HandlerEvent } from '../function'
+
+type HandlerWithoutResponse = (event: HandlerEvent, context: HandlerContext) => Promise<void>
+
+/**
+ * Declares a function to run on a cron schedule.
+ * Not reachable via HTTP.
+ *
+ * @example
+ * ```
+ * export const handler = cron("5 4 * * *", async () => {
+ *   // ...
+ * })
+ * ```
+ *
+ * @param schedule expressed as cron string. see https://crontab.guru for help.
+ * @param handler
+ * @see https://docs.netlify.com/functions/<schedule>
+ * @tutorial https://announcement-blogpost
+ */
+const schedule =
+  (cron: string, handler: HandlerWithoutResponse): Handler =>
```

Comment:
`cron` seems like a good candidate for a [template literal type](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html)

Reply:
was also thinking that! I didn't include it in the first draft since it would require all consumers to use a TypeScript version that supports template literal types (introduced in 4.1). I couldn't find any distribution information to quantify this, though ... Do you think that's a good tradeoff to make?

Reply:
Do we have a policy for minimum TS version? I'd think a year-old one would be ok

Reply:
I tried translating [this EBNF](https://docs.atlassian.com/caesium-parent/1.3.0/apidocs/com/atlassian/scheduler/caesium/cron/parser/CronExpressionParser.html) to a template literal string type.

Resulted in this TypeScript:

<details>

<p>

```typescript
// adapted from this ebnf:
// https://docs.atlassian.com/caesium-parent/1.3.0/apidocs/com/atlassian/scheduler/caesium/cron/parser/CronExpressionParser.html

type NUMBER = 'NUMBER'
type NAME = 'NUMBER'

type SlashInterval = `/${NUMBER}`
type NumberRange = `${NUMBER}-${NUMBER}${SlashInterval | ''}`
type NameRange = `${NAME}-${NAME}${SlashInterval | ''}`
type NumberExpression = `${NumberRange}${SlashInterval | ''}` | `${NUMBER}${SlashInterval}`
type NameExpression = NameRange | `${NAME}${SlashInterval | ''}`

type AsteriskExpression = `*${SlashInterval | ''}`

type SimpleExpression = NameExpression | NumberExpression | AsteriskExpression | SlashInterval // what is (* implied asterisk *) ?

type SimpleField =
  | SimpleExpression
  | `${SimpleExpression},${SimpleExpression}`
  | `${SimpleExpression},${SimpleExpression},${SimpleExpression}`
  | `${SimpleExpression},${SimpleExpression},${SimpleExpression},${SimpleExpression}`

type SpecialDomLastOffsetNumber = NUMBER | `${NUMBER}W`
type SpecialDomLastOffset = SpecialDomLastOffsetNumber
type SpecialDomLastWeekday = 'W'

type SpecialDomLast = `L-${SpecialDomLastOffset}` | `L${SpecialDomLastWeekday}` | 'L'

type SpecialDomWeekday = 'W'

type SpecialDomField = SpecialDomLast | SpecialDomWeekday

type SecondField = SimpleField
type MinuteField = SimpleField
type HourField = SimpleField
type DomField = SpecialDomField | SimpleField
type MonthField = SimpleField
type YearField = '' | SimpleField | `${SimpleField} ${string}`

type SecondMinuteHour = `${SecondField} ${MinuteField} ${HourField}`
type QmMonthDow = `? ${MonthField} ${DowField}`

type DomMonthQm = `${DomField} ${Monthfield} ?`
type DomMonthDow = QmMonthDow | DomMonthQm

type CronExpressionWithoutYear = `${SecondMinuteHour}${DomMonthDow}`
export type CronExpression = CronExpressionWithoutYear | `${CronExpressionWithoutYear} ${YearField}`
```

</p>

</details>

Sadly, that seems to be complex of an expression for TypeScript :(

```bash
❯ npm run build

> @netlify/functions@0.10.0 build
> tsc

src/lib/cron.ts:40:25 - error TS2590: Expression produces a union type that is too complex to represent.

40 type SecondMinuteHour = `${SecondField} ${MinuteField} ${HourField}`
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/lib/cron.ts:41:38 - error TS2552: Cannot find name 'DowField'. Did you mean 'DomField'?

41 type QmMonthDow = `? ${MonthField} ${DowField}`
                                        ~~~~~~~~

src/lib/cron.ts:41:38 - error TS4081: Exported type alias 'QmMonthDow' has or is using private name 'DowField'.

41 type QmMonthDow = `? ${MonthField} ${DowField}`
                                        ~~~~~~~~

src/lib/cron.ts:43:34 - error TS2552: Cannot find name 'Monthfield'. Did you mean 'MonthField'?

43 type DomMonthQm = `${DomField} ${Monthfield} ?`
                                    ~~~~~~~~~~

src/lib/cron.ts:43:34 - error TS4081: Exported type alias 'DomMonthQm' has or is using private name 'Monthfield'.

43 type DomMonthQm = `${DomField} ${Monthfield} ?`
                                    ~~~~~~~~~~

src/lib/cron.ts:47:58 - error TS2590: Expression produces a union type that is too complex to represent.

47 export type CronExpression = CronExpressionWithoutYear | `${CronExpressionWithoutYear} ${YearField}`
                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 6 errors.
```